src/hotspot/os/posix/os_posix.cpp

```diff
@@ -711,7 +711,16 @@ void* os::get_default_process_handle() {
 }
 
 void* os::dll_lookup(void* handle, const char* name) {
-  return dlsym(handle, name);
+  ::dlerror(); // Clear any previous error
+  void* ret = ::dlsym(handle, name);
+  if (ret == nullptr) {
+    const char* tmp = ::dlerror();
+    // It is possible that we found a NULL symbol, hence no error.
+    if (tmp != nullptr) {
+      log_debug(os)("Symbol %s not found in dll: %s", name, tmp);
```

Comment:
Would log_warning be more suitable?

Reply:
The reason why they are not warning is that we have code that is just checking if a function is available.
E.g. this happens like 40 times for math libs we try load and we don't want 40 warnings on every boot when that.

Reply:
Make sense, thanks for explanation.